pkg/engine/internal/planner/logical/planner.go

```diff
@@ -252,6 +256,13 @@ func buildPlanForSampleQuery(e syntax.SampleExpr, params logql.Params) (*Builder
 			}
 
 			rangeInterval = e.Left.Interval
+
+			// Check for unwrap in the LogRangeExpr
+			if e.Left.Unwrap != nil {
+				unwrapIdentifier = e.Left.Unwrap.Identifier
+				unwrapOperation = e.Left.Unwrap.Operation
```

Comment:
IMO we should be validating that we recognize and support whatever `e.Left.Unwrap.Operation` is and return errUnimplmeneted otherwise.